### PR TITLE
Different approach to preventing userland from emitting meta events

### LINKED ISSUE #97: Regresssion in PR#96
The https://github.com/sindresorhus/emittery/pull/96 introduces a regression when emittery is used as a sub-class. Lemme share a code sample.

```ts
const Emittery = require("emittery");

class MyEmitter extends Emittery {
  async emit(eventName, data) {
  	return super.emit(eventName, data)
  }
}

new MyEmitter().on('foo', () => {})
```

Adding a listener `foo` will raise an exception from this line https://github.com/sindresorhus/emittery/blob/main/index.js#L262, since it is calling `this.emit` with three arguments.

Now, I can update the `emit` method on my class to accept the 3rd argument and pass it to `super.emit`. However, then TypeScript complains that the `emit` method takes only two arguments.

### PR DESCRIPTION
**Pull request by lukehorvat**
Fixes #97. I implemented @thetutlage's solution [here](https://github.com/sindresorhus/emittery/issues/97#issuecomment-1129591450).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -157,6 +151,14 @@ function defaultMethodNamesOrAssert(methodNames) {
 
 const isMetaEvent = eventName => eventName === listenerAdded || eventName === listenerRemoved;
 
+async function emitMetaEvent(emitter, eventName, eventData) {
+	if (isMetaEvent(eventName)) {
+		canEmitMetaEvents = true;
+		await emitter.emit(eventName, eventData);
```
Comment (lukehorvat) [inline index.js:157]: What should we do if this throws an error? `canEmitMetaEvents` won't be reverted to `false`...

File: index.js [hunk 2]
```diff
@@ -157,6 +151,14 @@ function defaultMethodNamesOrAssert(methodNames) {
 
 const isMetaEvent = eventName => eventName === listenerAdded || eventName === listenerRemoved;
 
+function emitMetaEvent(emitter, eventName, eventData) {
+	if (isMetaEvent(eventName)) {
+		canEmitMetaEvents = true;
+		emitter.emit(eventName, eventData);
```
Comment (lukehorvat) [inline index.js:157]: What should we do if this throws an error? `canEmitMetaEvents` won't be reverted to false...

Comment (sindresorhus) [reply index.js:157]: Use try/finally? And set it to `false` in the `finally` block.

**Comment by sindresorhus** [comment]
@lukehorvat Bump :)

**Comment by lukehorvat** [comment]
Sorry, completely forgot to follow up on this... 🤦🏻‍♂️

I pushed a commit adding a `finally` block to ensure `canEmitMetaEvents` always gets reverted to `false`. So I think this should be ready to go now?

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**